CMakeLists.txt

```diff
@@ -24,6 +24,58 @@ endif()
 
 project (P4C)
 
+# Minimal lint-only configuration
+if (CMAKE_LINT_ONLY)
```

Comment:
This isn't really what I have been looking for. This just copies what is already there in the Linters file but without all the options. 

You should be able to pass linting with with `cmake .. -DCMAKE_LINT_ONLY` and also use the functions present in 
https://github.com/p4lang/p4c/blob/main/cmake/Linters.cmake.

Reply:
This would mean you should make the changes in the back ends and around the checks that fail in CMake.